source/extensions/omni.isaac.lab_tasks/test/test_environment_determinism.py

```diff
@@ -0,0 +1,153 @@
+
+                        # check everything is as expected
+                        # -- rewards should be the same
+                        torch.testing.assert_close(rew_1, rew_2)
+                        # -- observations should be the same
+                        for key in obs_1.keys():
+                            torch.testing.assert_close(obs_1[key], obs_2[key])
+
+    def test_locomotion_env_determinism(self):
+        """Check deterministic environment creation for locomotion."""
+        for task_name in [
+            "Isaac-Ant-v0",
+            "Isaac-Velocity-Flat-Anymal-C-v0",
+            "Isaac-Velocity-Rough-Anymal-C-v0",
+            "Isaac-Velocity-Flat-H1-v0",
+        ]:
+            for device in ["cuda", "cpu"]:
+                for seed in [25, 8001]:
+                    with self.subTest(task_name=task_name, device=device, seed=seed):
+                        # fix number of steps
+                        num_envs = 128
+                        num_steps = 600
+                        # call function to create and step the environment
+                        obs_1, rew_1 = self._obtain_transition_tuples(task_name, seed, num_envs, device, num_steps)
+                        obs_2, rew_2 = self._obtain_transition_tuples(task_name, seed, num_envs, device, num_steps)
+
+                        # check everything is as expected
+                        # -- rewards should be the same
+                        torch.testing.assert_close(rew_1, rew_2)
+                        # -- observations should be the same
+                        for key in obs_1.keys():
+                            torch.testing.assert_close(obs_1[key], obs_2[key])
+
+    def test_dextrous_env_determinism(self):
+        """Check deterministic environment creation for dextrous manipulation."""
+        for task_name in [
+            "Isaac-Repose-Cube-Allegro-v0",
+            "Isaac-Repose-Cube-Allegro-Direct-v0",
+        ]:
+            for device in ["cuda", "cpu"]:
+                for seed in [25, 8001]:
+                    with self.subTest(task_name=task_name, device=device, seed=seed):
+                        # fix number of steps
+                        num_envs = 128
+                        num_steps = 600
+                        # call function to create and step the environment
+                        obs_1, rew_1 = self._obtain_transition_tuples(task_name, seed, num_envs, device, num_steps)
+                        obs_2, rew_2 = self._obtain_transition_tuples(task_name, seed, num_envs, device, num_steps)
+
+                        # check everything is as expected
+                        # -- rewards should be the same
+                        torch.testing.assert_close(rew_1, rew_2)
+                        # -- observations should be the same
+                        for key in obs_1.keys():
+                            torch.testing.assert_close(obs_1[key], obs_2[key])
```

Comment:
I like the different task test cases, but do we have reason to believe there would be a difference in the determinism with the different tasks? 

Also, you could do them all in a single test case by adding another for loop which will reduce the duplicated code a ton

Reply:
Please check the later comment. Fixed the test to reduce code duplication.

Reply:
I meant having a single test case, but this works too!